flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/operator/CEPOperatorUtils.java

```diff
@@ -127,16 +134,36 @@
 					keySerializer,
 					nfaFactory));
 		} else {
-			patternStream = inputStream.transform(
+
+			KeySelector<T, Byte> keySelector = new NullByteKeySelector<>();
+			TypeSerializer<Byte> keySerializer = ByteSerializer.INSTANCE;
+
+			patternStream = inputStream.keyBy(new NullByteKeySelector<T>()).transform(
 				"TimeoutCEPPatternOperator",
 				eitherTypeInformation,
-				new TimeoutCEPPatternOperator<>(
+				new TimeoutKeyedCEPPatternOperator<>(
 					inputSerializer,
 					isProcessingTime,
+					keySelector,
+					keySerializer,
 					nfaFactory
 				)).forceNonParallel();
 		}
 
 		return patternStream;
 	}
+
+	/**
+	 * Used as dummy KeySelector to allow using WindowOperator for Non-Keyed Windows.
+	 * @param <T>
+	 */
+	protected static class NullByteKeySelector<T> implements KeySelector<T, Byte> {
```

Comment:
Duplicate code. Already exists in `AllWindowedStream`. Better to refactor the existing code to make it usable here.